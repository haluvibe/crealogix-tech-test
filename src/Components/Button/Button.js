import React from "react";
import PropTypes from "prop-types";

const Button = ({ handleClick, value }) => (
  <button onClick={() => handleClick()}>{value}</button>
);

Button.propTypes = {
  handleClick: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired
};

export default Button;
