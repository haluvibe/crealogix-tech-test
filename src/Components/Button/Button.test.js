import React from "react";
import { shallow } from "enzyme";
import Button from "./Button";

const handleClick = jest.fn();
const value = "Mock Value";

const mockProps = {
  handleClick,
  value
};

const setUp = (props = mockProps) => {
  const wrapper = shallow(<Button {...props} />);
  return wrapper;
};

describe("Button", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setUp();
  });

  it("should render correctly without errors", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Should include the text Mock Value", () => {
    expect(wrapper.text()).toEqual(value);
  });

  it("button is clickable and invokes the handleClick prop", () => {
    wrapper.find("button").simulate("click");
    expect(handleClick).toHaveBeenCalled();
  });
});
