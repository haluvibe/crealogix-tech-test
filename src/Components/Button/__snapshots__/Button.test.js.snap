// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly without errors 1`] = `
<button
  onClick={[Function]}
>
  Mock Value
</button>
`;
